checker/tests/nullness/java8/Issue579.java

```diff
@@ -18,7 +18,7 @@ public int compare(T a, T b) {
 
     @Override
     public Comparator<T> thenComparing(Comparator<? super T> other) {
-        // :: error: (argument.type.incompatible) :: warning: (known.nonnull)
+        // :: warning: (known.nonnull)
```

Comment:
The error that I get here originally:

```
checker/tests/nullness/java8/Issue579.java:22: error: [argument.type.incompatible] incompatible types in argument.
        return new Issue570<>(real == null ? other : real.thenComparing(other));
                                           ^
  found   : @Initialized @NonNull Comparator<capture#196 of ? super T extends @Initialized @Nullable Object super T extends @Initialized @Nullable Object>
  required: @Initialized @NonNull Comparator<? extends @Initialized @Nullable Object super T extends @Initialized @Nullable Object>
1 error
```

Was this really an expected error or was this a false positive?